tests/test_quorum.py

```diff
@@ -0,0 +1,122 @@
+import unittest
+
+from patroni.quorum import QuorumStateResolver, QuorumError
+
+class QuorumTest(unittest.TestCase):
+    def test_1111(self):
+        state = 1, set("a"), 1, set("a")
+        self.assertEqual(list(QuorumStateResolver(*state, active=set("ab"), sync_wanted=3)), [
+            ('sync', 2, set('ab')),
+            ('quorum', 1, set('ab')),
+        ])
+        self.assertEqual(list(QuorumStateResolver(*state, active=set("abcde"), sync_wanted=3)), [
+            ('sync', 3, set('abc')),
+            ('quorum', 3, set('abcde')),
+            ('sync', 3, set('abcde')),
+        ])
+
+    def test_1222(self):
+        """2 node cluster"""
+        state = 1, set("ab"), 2, set("ab")
+        # Active set matches state
+        self.assertEqual(list(QuorumStateResolver(*state, active=set("ab"), sync_wanted=3)), [
+        ])
+        # Add node by increasing quorum
```

Comment:
I find this description a bit too short , better would be to have sth like
> add a sync standby without increasing the replication factor; must increase quorum to avoid losing commits on failover